src/blog/2024/02/postgresql-with-node-red.md

```diff
@@ -0,0 +1,233 @@
+
+## Using PostgreSQL with Node-RED
+
+The first thing we need to do to get things started is to install the PostgreSQL custom node and gain an understanding of PostgreSQL configuration details.
+
+1. Install `node-red-contrib-postgresql` by the pallet manager. You can choose other nodes too, but we chose this node because it is part of the [certified node catalogue by FlowFuse](https://flowfuse.com/certified-nodes/) which assures that the node is robust, secure, and developed with high-quality.
+2. Before connecting to your PostgreSQL database, ensure you have the following information ready and environment variables set up as discussed below in `Adding environment variable` section:
+
+- Host: IP address or hostname of your PostgreSQL server.
+- Port: By default, PostgreSQL uses port 5432. Ensure this matches your PostgreSQL server configuration.
+- Database: The name of the PostgreSQL database you want to connect to.
+- User: username with the necessary privileges to access the specified database.
+- Password: corresponding password for the username.
+
+3. Drag the PostgreSQL node onto the canvas, click on that node and click on the edit icon next to the server input field to configure it.
+
+!["Configuring PostgreSQL"](./images/postgresql_with_node-red_pgconfig1.png "Configuring PostgreSQL")
+
+!["Configuring PostgreSQL"](./images/postgresql_with_node-red_pgconfig2.png "Configuring PostgreSQL")
+# Adding environment variables 
+
+We have discussed many times in previous blogs that using environment variables is a best practice that prevents revealing confidential configuration details such as API keys, passwords, secret keys, etc directly in the flow. Therefore, In this section, we will add environment variables for PostgreSQL configuration details, for more details see  [Using Environment Variables in Node-RED](https://flowfuse.com/blog/2023/02/environment-variables-in-node-red/).
+
+1. Navigate to the instance's setting and then go to the environment section.
+2. Click on the `add variable` button and add variables for each of the configuration data that we discussed in the above section.
+3. Click on the save button and restart the instance by clicking on the top right `Action` button and selecting the restart option.
+
+!["Adding environment variables"](./images/postgresql_with_nodred_environment_variable.png "Adding environment variables")
+
+# Creating Table
+
+In this section, we will create a table in our database to store product data.
+
+1. Drag an Inject node onto canvas, and keep it unchanged.
+2. Click on the PostgreSQL node that we have added previously, and paste the following SQL command into the query input field. (I have added comments for your understanding of SQL commands)
+
+```sql
+-- Create a table named product_data if it does not already exist
+CREATE TABLE IF NOT EXISTS product_data (
+  -- Define a column named id as a bigint type, which serves as the primary key
+  id bigint PRIMARY KEY,
```

Comment:
You don't need a bigint, no need to waste the bytes. You're not going to have 2 billion unique rows in this table so let's not overcomplicate things. 

Further, you're looking to auto increment this later, so you're actually looking for the SERIAL data type:

```suggestion
  id SERIAL PRIMARY KEY,
```